### PR TITLE
[WFLY-6854] Hibernate513

### PR DESCRIPTION
**Pull request by gbadner**
Incorporates https://github.com/wildfly/wildfly/pull/9257 for WFLY-6854, but updated to use Hibernate 5.1.3-SNAPSHOT

Update: I've added commit to use 5.1.3.Final now that it is released.

### CODE REVIEW

**Comment by scottmarlow** [comment]
Retest this please

**Comment by scottmarlow** [comment]
@kabir should the CI kick in automatically or do we need to manually start the testing?  I tried saying the magic "please" word but maybe I should say it a few more times. :)  Rest this please, please, please!

**Comment by luck3y** [comment]
this is ok to test

**Comment by scottmarlow** [comment]
We seemed to get a "Execution timeout (new);​ tests passed:​ 3599,​ ignored:​ 171 ", Retest this please.

**Comment by scottmarlow** [comment]
Retest this please

**Comment by gbadner** [comment]
this is ok to test

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java [hunk 1]
```diff
@@ -81,6 +81,7 @@ public void setupConfig() {
             Configuration configuration = new Configuration().setProperty(AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS,
                     "true");
             configuration.getProperties().put(AvailableSettings.JTA_PLATFORM, JBossAppServerJtaPlatform.class);
+            configuration.getProperties().put(AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY, "jta");
```
Comment (kabir) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java]: Is this a breaking change? Why do they need this now? In the worst case I guess they can document it

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/SFSB.java [hunk 2]
```diff
@@ -72,9 +72,9 @@ public void setupConfig() {
             //System.out.println("setupConfig:  Current dir = " + (new File(".")).getCanonicalPath());
 
             // prepare the configuration
-            Configuration configuration = new Configuration().setProperty(AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS,
-                    "true");
+            Configuration configuration = new Configuration().setProperty(AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS, "true");
             configuration.getProperties().put(AvailableSettings.JTA_PLATFORM, JBossAppServerJtaPlatform.class);
+            configuration.getProperties().put(AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY, "jta");
```
Comment (kabir) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/SFSB.java]: Why is this needed now?

**Review by kabir — COMMENTED** [review]
We also need the downstream PR before this can be merged. I will ask around a bit about the new property needed

**Comment by kabir** [comment]
Superseded by #9418 (I have put my comments from this PR in there)
